tests/snapshotter.spec.ts

```diff
@@ -211,14 +231,32 @@ it.describe('snapshots', () => {
   });
 });
 
+
+async function waitUntilPass(fn: (counter: number) => Promise<unknown>, sleep = 250) {
+  let counter = 0;
+  for (; ; ++counter) {
+    let err: unknown;
+    try {
+      await fn(counter);
+    } catch (e) {
```

Comment:
I don't really like that we rely on the exception - who knows what else could throw in this function? Can we do `captureSnapshotWithFrames` helper that would only do the frames trickery?

Reply:
Is `captureSnapshotWithFrames` like following?

```typescript
async function captureSnapshotWithFrames(snapshotter: InMemorySnapshotter, page: Page, toImpl:(rpcObject: any) => any) {
  for (let counter = 0; ; ++counter) {
    const snapshot = await snapshotter.captureSnapshot(toImpl(page), 'snapshot' + counter);
    const text = distillSnapshot(snapshot).replace(/frame@[^"]+["]/, '<id>"');
    if (text === '<IFRAME __playwright_src__=\"/snapshot/<id>\"></IFRAME>')
      break;
    await page.waitForTimeout(250);
  }
}
```

Reply:
Yeah, something like that. Or you can just duplicate this code in both tests, that's totally fine and probably easier than doing some helpers.